service/topology_coordinator.cc

```diff
@@ -959,24 +955,27 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
                 replica::tablet_mutation_builder(guard.write_timestamp(), table_id)
                     .set_resize_decision(std::move(resize_decision))
                     .build());
+    }
+
+    future<> generate_migration_updates(std::vector<canonical_mutation>& out, const group0_guard& guard, const migration_plan& plan) {
+        for (const tablet_migration_info& mig : plan.migrations()) {
+            co_await coroutine::maybe_yield();
+            generate_migration_update(out, guard, mig);
         }
 
-        // FIXME: Finalize split requests when exiting the tablet migration track.
-        for (auto table_id : plan.resize_plan().finalize_resize) {
-            auto& tmap = get_token_metadata_ptr()->tablets().get_tablet_map(table_id);
-            // Only finalize split request if there's no ongoing transition or new ones for a given table.
-            if (tmap.transitions().size() > 0 || new_transitions.contains(table_id)) {
-                continue;
+        for (auto [table_id, resize_decision] : plan.resize_plan().resize) {
+            co_await coroutine::maybe_yield();
+            generate_resize_update(out, guard, table_id, resize_decision);
+
+            // If split decision was revoked, let's update the state to prevent the split from being
+            // unnecessarily finalized, which might in turn require a merge.
+            if (!resize_decision.split_or_merge() && _tables_to_finalize_split.contains(table_id)) {
+                _tables_to_finalize_split.erase(table_id);
```

Comment:
Why do we need to keep state in _tables_to_finalize_split?

This function shouldn't have this logic, it should just convert the plan to mutations. plan-maker should make those decisions.

Reply:
you're right, I think we should move it all to maybe_start_tablet_split_finalize().